addons/mail/static/src/models/activity_box_view/activity_box_view.js

```diff
@@ -0,0 +1,48 @@
+/** @odoo-module **/
+
+import { registerNewModel } from '@mail/model/model_core';
+import { attr, one2one } from '@mail/model/model_field';
+import { clear, link } from '@mail/model/model_field_command';
+
+function factory(dependencies) {
+
+    class ActivityBoxView extends dependencies['mail.model'] {
+
+        /**
+         * @override
+         */
+        _created() {
+            // Bind necessary until OWL supports arrow function in handlers: https://github.com/odoo/owl/issues/876
+            this.onClickActivityBoxTitle = this.onClickActivityBoxTitle.bind(this);
+        }
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * Handles click on activity box title.
+         */
+        onClickActivityBoxTitle() {
+            this.update({ isActivityBoxVisible: !this.isActivityBoxVisible });
+        }
+
+    }
+
+    ActivityBoxView.fields = {
+        chatter: one2one('mail.chatter', {
+            inverse: 'activityBoxView',
+            required: true,
+            readonly: true,
+        }),
+        isActivityBoxVisible: attr({
```

Comment:
As you have diff on 100% of the lines using this field, I would take the opportunity to rename it to something better.
As a first idea, `isActivityListVisible` would already be better, as it is the name of a part of the box that we control with this variable.
Another idea would be to have some kind of "display mode" variable with collapsed/summary and full/list as options. Note that the existence of such variable is a hint that maybe we should have a split into 2 separate views (definitely not to be done in this PR though).